### PR TITLE
Add missing unit tests for segment loading in historicals

### PR DESCRIPTION
**Pull request by jihoonson**
### Description

This PR adds missing unit tests for the changes made in https://github.com/apache/druid/pull/10650.

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

<hr>

### CODE REVIEW

**Comment by zhangyue19921010** [comment]
Thanks for adding this UT. LTGM :)

File: server/src/test/java/org/apache/druid/server/coordination/SegmentLoadDropHandlerTest.java [hunk 1]
```diff
@@ -29,22 +29,28 @@
...
 import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.EmittingLogger;
 import org.apache.druid.segment.IndexIO;
 import org.apache.druid.segment.TestHelper;
 import org.apache.druid.segment.loading.CacheTestSegmentLoader;
 import org.apache.druid.segment.loading.SegmentLoaderConfig;
 import org.apache.druid.segment.loading.StorageLocationConfig;
 import org.apache.druid.server.SegmentManager;
+import org.apache.druid.server.coordination.SegmentLoadDropHandler.DataSegmentChangeRequestAndStatus;
+import org.apache.druid.server.coordination.SegmentLoadDropHandler.Status;
```
Comment (zhangyue19921010) [inline server/src/test/java/org/apache/druid/server/coordination/SegmentLoadDropHandlerTest.java:40]: nit: Need to remove this unused import.

**Review by zhangyue19921010 — APPROVED** [review]

File: server/src/test/java/org/apache/druid/server/coordination/SegmentLoadDropHandlerTest.java (hunk 1 above)
Comment (jihoonson) [reply server/src/test/java/org/apache/druid/server/coordination/SegmentLoadDropHandlerTest.java]: Oops, thanks!

**Review by a2l007 — APPROVED** [review]
👍

**Review by jon-wei — APPROVED** [review]

**Pull request merged**